src/main/java/edu/hm/hafner/analysis/parser/JsonParser.java

```diff
@@ -0,0 +1,179 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.LineRange;
+import edu.hm.hafner.analysis.LineRangeList;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * Base class for JSON parsers.
+ *
+ * @author Jeremie Bresson
+ */
+public class JsonParser extends IssueParser {
+    private static final long serialVersionUID = 1349282064371959197L;
+
+    /** Key for additionalProperties. */
+    private static final String ADDITIONAL_PROPERTIES = "additionalProperties";
+    /** Key for category. */
+    private static final String CATEGORY = "category";
+    /** Key for columnEnd. */
+    private static final String COLUMN_END = "columnEnd";
+    /** Key for columnStart. */
+    private static final String COLUMN_START = "columnStart";
+    /** Key for description. */
+    private static final String DESCRIPTION = "description";
+    /** Key for directory. */
+    private static final String DIRECTORY = "directory";
+    /** Key for end in lineRange. */
+    private static final String END = "end";
+    /** Key for fileName. */
+    private static final String FILE_NAME = "fileName";
+    /** Key for fingerprint. */
+    private static final String FINGERPRINT = "fingerprint";
+    /** Key for id. */
+    private static final String ID = "id";
+    /** Key for lineEnd. */
+    private static final String LINE_END = "lineEnd";
+    /** Key for lineRanges. */
+    private static final String LINE_RANGES = "lineRanges";
+    /** Key for lineStart. */
+    private static final String LINE_START = "lineStart";
+    /** Key for message. */
+    private static final String MESSAGE = "message";
+    /** Key for moduleName. */
+    private static final String MODULE_NAME = "moduleName";
+    /** Key for origin. */
+    private static final String ORIGIN = "origin";
+    /** Key for packageName. */
+    private static final String PACKAGE_NAME = "packageName";
+    /** Key for reference. */
+    private static final String REFERENCE = "reference";
+    /** Key for severity. */
+    private static final String SEVERITY = "severity";
+    /** Key for start in lineRange. */
+    private static final String START = "start";
+    /** Key for type. */
+    private static final String TYPE = "type";
+
+    @Override
+    public boolean accepts(final ReaderFactory readerFactory) {
+        return true;
```

Comment:
check the extension for `.json`.

Reply:
Are you sure about this?
The file is not a proper JSON file but a log file containing the entries serialised as JSON on each line.

Reply:
Ah, I missed that. 

The reason for the method is that if we add our parser to the [same tool](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/java/io/jenkins/plugins/analysis/warnings/WarningsPlugin.java) in the warnings plugin, then we get no error messages if the parser will receive an XML file. It is similar to the [Eclipse configuration](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/java/io/jenkins/plugins/analysis/warnings/Eclipse.java). 

In order to get this still running we can use the expression:
```
return !readerFactory.getFileName().endsWith(".xml");
```